bundles/org.openhab.binding.bluelink/src/main/java/org/openhab/binding/bluelink/internal/handler/BluelinkAccountHandler.java

```diff
@@ -0,0 +1,160 @@
+ */
+@NonNullByDefault
+public class BluelinkAccountHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BluelinkAccountHandler.class);
+
+    private final HttpClient httpClient;
+    private final TimeZoneProvider timeZoneProvider;
+    private final ConcurrentMap<String, VehicleInfo> vehicles = new ConcurrentHashMap<>(1);
+
+    private volatile @Nullable BluelinkApi api;
+
+    public BluelinkAccountHandler(final Bridge bridge, final HttpClient httpClient,
+            final TimeZoneProvider timeZoneProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Bluelink account handler");
+
+        final BluelinkAccountConfiguration config = getConfigAs(BluelinkAccountConfiguration.class);
+        final String username = config.username;
+        final String password = config.password;
+        if (username == null || username.isBlank() || password == null || password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Username and password are required");
+            return;
+        }
+
+        final String configuredBaseUrl = config.apiBaseUrl;
+        final String baseUrl = configuredBaseUrl != null ? configuredBaseUrl : API_ENDPOINT;
+        this.api = new BluelinkApi(httpClient, baseUrl, timeZoneProvider, username, password, config.pin);
+        scheduler.execute(this::login);
+    }
+
+    private void login() {
+        final BluelinkApi bluelinkApi = api;
+        if (bluelinkApi == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "API not initialized");
+            return;
+        }
+
+        try {
+            if (bluelinkApi.login()) {
+                logger.debug("Bluelink login successful");
+                loadEnrolledVehicles();
+                logger.debug("Found {} vehicles", vehicles.size());
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Login failed");
+            }
+        } catch (final BluelinkApiException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Bluelink account handler");
+        api = null;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // Bridge has no channels to handle
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return List.of(BluelinkVehicleDiscoveryService.class);
+    }
+
+    public @Nullable BluelinkApi getApi() {
+        return api;
+    }
```

Comment:
By design, it is a bad practice to expose the api. This account handler is responsible for interacting to the API.
Simplified:
vehiclehandler <-> accounthandler <-> apicontroller

The accounthandler should have a clear interface for child handlers and can act as some kind of proxy